frontend/src/views/student/tournament/MyTournamentsView.vue

```diff
@@ -0,0 +1,348 @@
+  createTournamentDialog: boolean = false;
+  editTournamentDialog: boolean = false;
+  search: string = '';
+  headers: object = [
+    {
+      text: 'Actions',
+      value: 'action',
+      align: 'left',
+      sortable: false,
+      width: '20%'
+    },
+    {
+      text: 'Course Acronym',
+      value: 'courseAcronym',
+      align: 'center',
+      width: '10%'
+    },
+    { text: 'Tournament Number', value: 'id', align: 'center', width: '10%' },
+    {
+      text: 'Topics',
+      value: 'topics',
+      align: 'center'
+    },
+    {
+      text: 'State',
+      value: 'isCanceled',
+      align: 'center',
+      width: '10%'
+    },
+    {
+      text: 'Privacy',
+      value: 'privateTournament',
+      align: 'center',
+      width: '10%'
+    },
+    {
+      text: 'Start / End Time',
+      value: 'times',
+      align: 'center',
+      width: '10%'
+    },
+    {
+      text: 'Number of Questions',
+      value: 'numberOfQuestions',
+      align: 'center',
+      width: '5%'
+    },
+    {
+      text: 'Enrolled',
+      value: 'enrolled',
+      align: 'center',
+      sortable: false,
+      width: '10%'
+    }
+  ];
+
+  async created() {
+    await this.$store.dispatch('loading');
+    try {
+      this.tournaments = await RemoteServices.getUserTournaments();
+      let closedTournaments = await RemoteServices.getClosedTournaments();
+      closedTournaments.map(t => {
+        if (t.id) this.closedTournamentsId.push(t.id);
+      });
+      this.tournaments.sort((a, b) => this.sortById(a, b));
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    await this.$store.dispatch('clearLoading');
+  }
+
+  sortById(a: Tournament, b: Tournament) {
+    if (a.id && b.id) return a.id > b.id ? 1 : -1;
+    else return 0;
+  }
+
+  openTournamentDashboard(tournament: Tournament) {
+    if (tournament) return '/student/tournament?id=' + tournament.id;
+  }
+
+  newTournament() {
+    this.currentTournament = new Tournament();
+    this.createTournamentDialog = true;
+  }
+
+  async onCreateTournament(tournament: Tournament) {
+    this.tournaments.unshift(tournament);
+    this.createTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  onCloseDialog() {
+    this.createTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  editTournament(tournamentToEdit: Tournament) {
+    this.currentTournament = tournamentToEdit;
+    this.editTournamentDialog = true;
+  }
+
+  async onEditTournament(tournament: Tournament) {
+    this.currentTournament = tournament;
+    try {
+      this.tournaments = await RemoteServices.getUserTournaments();
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    this.editTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  onCloseEditDialog() {
+    this.editTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  getStateColor(tournament: Tournament) {
```

Comment:
repeated code, shoud be in tournament...

Reply:
Done